cylc/flow/xtrigger_mgr.py

```diff
@@ -261,32 +259,61 @@ def _validate(
                 raise XtriggerConfigError(
                     label,
                     (
-                        f"xtrigger '{fname}' function definition contains "
-                        "reserved argument 'sequential' that has no "
-                        "boolean default"
+                        f"xtrigger '{fctx.func_name}' has a reserved argument"
+                        " 'sequential' with no boolean default"
                     )
                 )
             fctx.func_kwargs.setdefault('sequential', sequential_param.default)
 
         elif 'sequential' in fctx.func_kwargs:
-            # xtrig call marked as sequential; add 'sequential' arg to
-            # signature for validation
+            # xtrig marked as sequential, so add 'sequential' arg to signature
             sig = add_kwarg_to_sig(
                 sig, 'sequential', fctx.func_kwargs['sequential']
             )
 
         # Validate args and kwargs against the function signature
         try:
-            bound_args = sig.bind(
-                *fctx.func_args, **fctx.func_kwargs
-            )
+            bound_args = sig.bind(*fctx.func_args, **fctx.func_kwargs)
         except TypeError as exc:
-            raise XtriggerConfigError(label, f"{sig_str}: {exc}")
+            # try fname_backcompat
```

Comment:
I don't think this should be done on a general basis, it should only be done for `workflow_state` - I'm happy to write a PR against this one if you'd like

Reply:
This seems pretty safe to me - can you see it causing trouble somehow? 

Anyway, I don't mind if you want to tweak it to be workflow_state-specific - it's unlikely (if not impossible) to be more widely useful.

(I think we should keep my change to allow different function and module names though).

Reply:
The code will be more straightforward if we only handle the case of `workflow_state`

> (I think we should keep my change to allow different function and module names though).

I don't think this is used though?

Reply:
It's used for finding the "validate" function in the xtrigger module.